source/base/data_out_base.cc

```diff
@@ -6354,17 +6281,43 @@ namespace DataOutBase
                         const unsigned int d1 = 1;
                         const unsigned int d2 = n;
                         const unsigned int d3 = n * n;
+
+                        auto write_cell = [&flags, &out, &cells, d1, d2, d3](
+                                            const unsigned int start) {
```

Comment:
✔️

Reply:
I like this style but we now get some unused variable warnings without zlib. Could you add

```cpp
(void)d1;
(void)d2;
(void)d3;
```
to the other case or some other equivalent fix?